### PR TITLE
HPCC-20372 Removing subfiles action does nothing

### PR DESCRIPTION
**Pull request by miguelvazq**
Querying the store did nothing when refreshing. ESPLogicalFile handles
the refresh just fine.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by miguelvazq** [comment]
@GordonSmith Please review.

File: esp/src/eclwatch/SFDetailsWidget.js [hunk 1]
```diff
@@ -93,7 +87,6 @@ define([
             _onRemove: function (event) {
                 if (confirm(this.i18n.RemoveSubfiles2)) {
                     this.logicalFile.removeSubfiles(this.subfilesGrid.getSelected());
-                    this._onSubFileRefresh();
```
Comment (GordonSmith) [inline esp/src/eclwatch/SFDetailsWidget.js]: I think the following might be preferable (will need a test):
```
var context = this;
this.logicalFile.removeSubfiles(this.subfilesGrid.getSelected()).then(function() {
    context._onSubFileRefresh();
});
```

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/eclwatch/SFDetailsWidget.js (hunk 1 above)
Comment (GordonSmith) [reply esp/src/eclwatch/SFDetailsWidget.js]: IOW currently the refresh is happening before the remove has completed.

**Comment by miguelvazq** [comment]
@GordonSmith bump

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/eclwatch/SFDetailsWidget.js [hunk 2]
```diff
@@ -91,9 +85,11 @@ define([
                 }
             },
             _onRemove: function (event) {
+                var context = this;
                 if (confirm(this.i18n.RemoveSubfiles2)) {
-                    this.logicalFile.removeSubfiles(this.subfilesGrid.getSelected());
-                    this._onSubFileRefresh();
+                    this.logicalFile.removeSubfiles(this.subfilesGrid.getSelected()).then(function(){
+                        context._onSubFileRefresh();
```
[the remark this answers is not in the document]
Comment (miguelvazq) [reply esp/src/eclwatch/SFDetailsWidget.js]: Looks like my last commit did not get pushed

[the remark this answers is not in the document]
Comment (miguelvazq) [reply esp/src/eclwatch/SFDetailsWidget.js]: Tested with _onSubFileRefresh seems to just refresh grid with "Zero Rows". The setting of the query seems to zero out the grid results. The refresh ESPLogicalFile seems to do the job which is why I omitted the setting of the query.

**Comment by miguelvazq** [comment]
@GordonSmith Updated. Feel free to test

**Review by GordonSmith — APPROVED** [review]
Looks fine.

**Pull request merged**